File: carnivorecore/create-post.py
import json
import re
from datetime import date
from pathlib import Path

def slugify(text):
    return re.sub(r'[^a-z0-9]+', '-', text.lower()).strip('-')

title = input("Title: ")
slug = slugify(title)
excerpt = input("Excerpt: ")
today = date.today().isoformat()
id = int(date.today().strftime('%Y%m%d'))

post_md = f"""---
title: \"{title}\"
date: \"{today}\"
id: {id}
---

Write your post content here.
"""

post_path = Path("posts") / f"{slug}.md"
post_path.write_text(post_md)

index_path = Path("posts") / "index.json"
index_data = json.loads(index_path.read_text())
index_data.append({
    "slug": slug,
    "title": title,
    "date": today,
    "excerpt": excerpt
})
index_data.sort(key=lambda x: x['date'], reverse=True)
index_path.write_text(json.dumps(index_data, indent=2))

print(f"✅ Created posts/{slug}.md and updated index.json")

File: carnivorecore/create-post.js
const fs = require('fs');
const path = require('path');
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const ask = (q) => new Promise(resolve => rl.question(q, resolve));
const slugify = (text) => text.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/(^-|-$)/g, '');

(async () => {
  const title = await ask('Title: ');
  const slug = slugify(title);
  const date = new Date().toISOString().split('T')[0];
  const excerpt = await ask('Excerpt: ');

  const frontmatter = `---\ntitle: "${title}"\ndate: "${date}"\nid: ${Date.now()}\n---\n\nWrite your post content here.`;

  const postPath = path.join(__dirname, 'posts', `${slug}.md`);
  fs.writeFileSync(postPath, frontmatter);

  const indexPath = path.join(__dirname, 'posts', 'index.json');
  const indexData = JSON.parse(fs.readFileSync(indexPath));
  indexData.push({ slug, title, date, excerpt });
  indexData.sort((a, b) => new Date(b.date) - new Date(a.date));
  fs.writeFileSync(indexPath, JSON.stringify(indexData, null, 2));

  console.log(`✅ Created post at posts/${slug}.md`);
  rl.close();
})();

